activerecord/lib/active_record/errors.rb

```diff
@@ -29,6 +29,17 @@ class AssociationTypeMismatch < ActiveRecordError
 
   # Raised when unserialized object's type mismatches one specified for serializable field.
   class SerializationTypeMismatch < ActiveRecordError
+    attr_accessor :attribute
+
+    def initialize(expected_class, actual_class, value)
+      @expected_class = expected_class
+      @actual_class = actual_class
+      @value = value
+    end
+
+    def to_s
```

Comment:
Generally I'd expect `message` to hold this for an exception. Why are we using `to_s`?